src/core/image.js

```diff
@@ -535,6 +535,128 @@ var PDFImage = (function PDFImageClosure() {
       }
     },
 
+    /**
+     * Check whether resize image data or not
+     * @param {Number} comps Number of color components, 1 or 3 is supported.
+     * @param {Number} bpc Number of bits per component.
+     * @return {Boolean}
+     */
+    shallResizeImage: function PDFImage_shallResizeImage(comps, bpc) {
+      return (comps === 1 && (bpc === 1 || bpc === 8));
+    },
+
+    /**
+     * Resize black white image data
+     * @param {Uint8Array} imgData image data.
+     * @param {Number} scaleBits
+     * @author Ramsoft
```

Comment:
We do not use `@author` JSDoc annotations. Make sure that the committer of the commit is set correctly if you wish to include this in the version history and remove this line.